tests/test_addon.py

```diff
@@ -67,8 +71,8 @@ def _test_setting_changes(initial_settings, expected_settings):
                 "CLOSESPIDER_TIMEOUT_NO_ITEM": 600,
                 "DOWNLOADER_MIDDLEWARES": {
                     MaxRequestsPerSeedDownloaderMiddleware: 100,
-                    OffsiteMiddleware: None,
-                    AllowOffsiteMiddleware: 500,
+                    BUILTIN_OFFSITE_MIDDLEWARE_IMPORT_PATH: None,
+                    AllowOffsiteMiddleware: 50,
```

Comment:
As far as I can see from the code, this change in order (the offsite middleware will now come before the max-requests-per-seed middleware) is OK, probably even for the best. I hope it does not come back to bite me…

Reply:
:) It's interesting as all tests for the addon passed - might be this case was not covered :(

Reply:
The add-on was injecting the built-in middleware manually (the lines removed) before calling `_replace_builtin`, that’s why it was not caught by the tests.